### PR TITLE
triton-dehydrated#40 Sometimes challenge value can start with - which…

### PR DESCRIPTION
**Pull request by bahamat**
… severely confuses grep

### CODE REVIEW

File: cns-hook-util [hunk 1]
```diff
@@ -49,8 +49,7 @@ function waitfortxt {
         read -r -a values <<< "$txtval"
         for val in "${values[@]}"; do
                 while [ $count -lt 3 ]; do
-                        dig +short txt "_acme-challenge.${domain}" @${NAMESERVER:-8.8.8.8}
```
Comment (bahamat) [inline cns-hook-util:52]: Removed this spurious line

File: cns-hook-util [hunk 2]
```diff
@@ -49,8 +49,7 @@ function waitfortxt {
...
-                        if dig +short txt "_acme-challenge.${domain}" @${NAMESERVER:-8.8.8.8} | grep -w "$val" >/dev/null; then
```
Comment (bahamat) [inline cns-hook-util:53]: changed from `grep -w "$val"` to `fgrep -- "$val"`.

File: cns-hook-util [hunk 3]
```diff
@@ -49,8 +49,7 @@ function waitfortxt {
...
+                        if dig +short txt "_acme-challenge.${domain}" @${NAMESERVER:-8.8.8.8} | fgrep -- "$val" >/dev/null; then
```
Comment (bahamat) [inline cns-hook-util:52]: changed from `grep -w "$val"` to `fgrep -- "$val"`.

**Review by numericillustration — APPROVED** [review]
looks like `fgrep` and `grep -F` are equally portable since gnugrep has the same aliases

**Pull request merged**